src/shared/cpu-set-util.c

```diff
@@ -82,6 +82,53 @@ char *cpu_set_to_range_string(const CPUSet *set) {
         return TAKE_PTR(str) ?: strdup("");
 }
 
+char* cpu_set_to_mask_string(const CPUSet *a) {
+        _cleanup_free_ char *str = NULL;
+        size_t len = 0;
+        bool nz = 0;
+        int i, r;
```

Comment:
`i` should be of type `size_t`. After all you iterate through memory with it, and that's generally the type we use for that. Moreover, you use it to count to field `allocated`, which itself is `size_t`, hence please stick to the same type.

Reply:
also, move `size_t i` to innermost scope, i.e. to become `for (size_t i = …` below

Reply:
here, i should be signed, to exit the for loop `i >= 0`